### PR TITLE
Is slashable block rpc endpoint implementation

### LINKED ISSUE #5189: reintroduce slasher RPC server
in order to support validator redundancy slasher need to support external calls:
- [x] slasher grpc server #5190
- [x] implement IsSLashableAttestation endpoint #5209
- [ ] implement IsSLashableBlock endpoint #5765
- [x] retrieve and cache validators public keys #5220
- [x] get fork version and genesis validators root ethereumapi endpoint 
- [x] implement ethereumapi endpoint in beacon node
- [x] validate signature of rpc requests

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**
In order to enable validator protection slasher needs an endpoint that registered users can use to make sure that the validator is not going to be slashed by submitting their duties.

**Which issues(s) does this PR fix?**

Fixes #5189

### CODE REVIEW

File: beacon-chain/core/helpers/signing_root.go [hunk 1]
```diff
@@ -98,6 +98,29 @@ func VerifyBlockSigningRoot(blk *ethpb.BeaconBlock, pub []byte, signature []byte
...
+	publicKey, err := bls.PublicKeyFromBytes(pub)
+	if err != nil {
+		return errors.Wrap(err, "could not convert bytes to public key")
+	}
+	sig, err := bls.SignatureFromBytes(signature)
+	if err != nil {
+		return errors.Wrap(err, "could not convert bytes to signature")
+	}
+	root, err := signingRoot(func() ([32]byte, error) {
+		// utilize custom block hashing function
```
Comment (rauljordan) [inline beacon-chain/core/helpers/signing_root.go:112]:
```suggestion
```

File: slasher/rpc/server.go [hunk 2]
```diff
@@ -91,5 +91,37 @@ func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Indexed
 // IsSlashableBlock returns an proposer slashing if the block submitted
 // is a double proposal.
 func (ss *Server) IsSlashableBlock(ctx context.Context, req *ethpb.SignedBeaconBlockHeader) (*slashpb.ProposerSlashingResponse, error) {
-	return nil, errors.New("unimplemented")
+	ctx, span := trace.StartSpan(ctx, "detection.IsSlashableBlock")
+	defer span.End()
+
+	gvr, err := ss.beaconClient.GenesisValidatorsRoot(ctx)
```
Comment (rauljordan) [inline slasher/rpc/server.go:97]: This value should only be fetched once, any chance we can retrieve it somewhere else or cache future calls?

File: slasher/rpc/server.go [hunk 3]
```diff
@@ -91,5 +91,37 @@ func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Indexed
...
+	if err != nil {
+		return nil, err
+	}
+	blockEpoch := helpers.SlotToEpoch(req.Header.Slot)
```
Comment (rauljordan) [inline slasher/rpc/server.go:101]: Can we make slasher panic if we pass in a bad request? Maybe add a nil check

File: slasher/rpc/server.go [hunk 4]
```diff
@@ -91,5 +91,37 @@ func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Indexed
...
+	fork, err := p2putils.Fork(blockEpoch)
+	if err != nil {
+		return nil, err
+	}
+	domain, err := helpers.Domain(fork, blockEpoch, params.BeaconConfig().DomainBeaconProposer, gvr)
+	if err != nil {
+		return nil, err
+	}
+	pkMap, err := ss.beaconClient.FindOrGetPublicKeys(ctx, []uint64{req.Header.ProposerIndex})
```
Comment (rauljordan) [inline slasher/rpc/server.go:110]: This result can likely also use a caching-layer given there may be many repeat requests

Comment (shayzluf) [reply slasher/rpc/server.go:110]: This method is cached

File: slasher/rpc/server.go (hunk 2 above)
Comment (shayzluf) [reply slasher/rpc/server.go:97]: thanks, added

File: beacon-chain/core/helpers/signing_root.go (hunk 1 above)
Comment (shayzluf) [reply beacon-chain/core/helpers/signing_root.go:112]: removed, thanks

File: slasher/rpc/server.go (hunk 3 above)
Comment (shayzluf) [reply slasher/rpc/server.go:101]: added. thanks

**Pull request merged**